File: frontend/js/command/unit-types/camera.js
// Created by Matthew Valancy
// Copyright 2026 Valpatel Software LLC
// Licensed under AGPL-3.0 — see LICENSE for details.
import { UnitType } from './base.js';
import { registerType } from './registry.js';
import Sensor from './sensor.js';

class Camera extends UnitType {
    static typeId = 'camera';
    static displayName = 'Camera';
    static iconLetter = 'C';
    static visionRadius = 30;
    static cotType = 'a-f-G-E-S-E';
    static ambientRadius = 5;
    static coneRange = 30;
    static coneAngle = 60;
    static coneSweeps = true;
    static coneSweepRPM = 1;

    /** Reuses Sensor draw -- same FOV cone + body + lens. */
    static draw(ctx, scale, color) {
        Sensor.draw(ctx, scale, color);
    }
}

registerType(Camera);
export default Camera;

File: src/engine/units/sensors/camera.py
# Created by Matthew Valancy
# Copyright 2026 Valpatel Software LLC
# Licensed under AGPL-3.0 — see LICENSE for details.
from engine.units.base import CombatStats, MovementCategory, UnitType


class Camera(UnitType):
    type_id = "camera"
    display_name = "Camera"
    icon = "C"
    cot_type = "a-f-G-E-S-E"  # Friendly ground emplaced sensor
    category = MovementCategory.STATIONARY
    speed = 0.0
    drain_rate = 0.0
    vision_radius = 30.0
    ambient_radius = 5.0
    cone_range = 30.0
    cone_angle = 60.0
    cone_sweeps = True
    cone_sweep_rpm = 1.0
    placeable = False
    combat = CombatStats(
        health=50, max_health=50,
        weapon_range=0.0, weapon_cooldown=0.0, weapon_damage=0,
        is_combatant=False,
    )
